packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -926,15 +924,18 @@ export class LocalFluidDataStoreContextBase extends FluidDataStoreContext {
                 // If there is no isRootDataStore in the attributes blob, set it to true. This ensures that data
                 // stores in older documents are not garbage collected incorrectly. This may lead to additional
                 // roots in the document but they won't break.
-                this.isRootDataStore = this.isRootDataStore || (attributes.isRootDataStore ?? true);
+                this.isLocalRootDataStore = this.isLocalRootDataStore ||
```

Comment:
How should this be handled? I was thinking this.setRoot if attributes.isRootDataStore ?? true

Reply:
We should not be setting any state here. We should only be reading it. For local data stores, the state should be set during data store creation or when it is aliased and isRoot is called.

Reply:
Actually, that may not be true. Let me take a look at the code closely.

Reply:
Okay, so this has to go back to how it originally was. Basically if the root (or alias) state is already true, then use it, else use from attributes:
`this.isRootDataStore = this.isRootDataStore || (attributes.isRootDataStore ?? true);`

Reply:
Ok, because I wanted to make the `this.isRootDataStore` variable private and thus not accessible to the child, in this case the `LocalFluidDataStoreContextBase`, I'll do something like `if(this.isAliased || (attributes.isRootDataStore ?? true)) this.setRoot/setAliased`.

Reply:
Sounds good. Although if its already aliased, you don't need to set it as root or aliased again.